patternfly/patternfly-elements
# fix: pfe-accordion - fixing issues with accessibility when opening and closing panels (#1813)

Fixes #1813

Fixed issues with panels expanding when headers received focus on keyboard interactions.

### Preview
Link(s) to demo page(s) where this element can be viewed:
- [Link](https://deploy-preview-1856--patternfly-elements.netlify.app/components/accordion/) 


### What has changed and why
Added the appropriate aria attributes and tabindex attribute to accordion panels when they are collapsed and removed those attributes when the accordion panels are expanded. Also removed the `expand` method call in the keyboard events section to prevent accordion panels from opening when navigating the accordion with a keyboard.

### Testing instructions

  1. [Go to the accordion demo page](https://deploy-preview-1856--patternfly-elements.netlify.app/elements/pfe-accordion/demo)
  2. Activate an accordion header
  3. Expect the associated panel to *not* have an aria-hidden attribute and a tabindex attribute
  4. Deactivate the same accordion header
  5. Expect the associated panel to have an `aria-hidden="true"` attribute and a `tabindex="-1"` 


#### Browser requirements

Your component should work in all of the following environments:



### Ready-for-merge Checklist



### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**
